src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,165 @@
+-- bipartite graph by two maps of vertices into its adjacency sets. The two
+-- type arguments for this datatype stand for the identifiers of the vertices
+-- of each part.
+--
+--  __Note:__ even if the identifiers and their types for two vertices of
+-- different parts are equal, these vertices are considered to be different.
+-- See the examples for more details.
+data AdjacencyMap a b = BAM {
+    -- | The /adjacency map/ of the left part of the graph: each vertex is
+    -- associated with a set of its neighbours. Complexity: /O(1)/ time and
+    -- memory
+    --
+    -- @
+    -- leftToRight ('fromGraph' 'Algebra.Graph.empty')                                          == Map.'Map.empty'
+    -- leftToRight ('fromGraph' ('Algebra.Graph.vertex' (Left 1)))                              == Map.'Map.singleton' 1 Set.'Set.empty'
+    -- leftToRight ('fromGraph' ('Algebra.Graph.vertex' (Right 1)))                             == Map.'Map.empty'
+    -- leftToRight ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 2)))                      == Map.'Map.singleton' 1 (Set.'Set.singleton' 2)
+    -- leftToRight ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 1)))                      == Map.'Map.singleton' 1 (Set.'Set.singleton' 1)
+    -- leftToRight ('fromGraph' ('Algebra.Graph.edges' [(Left 1, Right 1), (Left 1, Right 2)])) == Map.'Map.singleton' 1 (Set.'Set.fromAscList' [1, 2])
+    -- @
+    leftToRight :: Map.Map a (Set.Set b),
+
+    -- | The inverse map for 'leftToRight'. Complexity: /O(1)/ time and memory.
+    --
+    -- @
+    -- rightToLeft ('fromGraph' 'Algebra.Graph.empty')                                          == Map.'Map.empty'
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.vertex' (Left 1)))                              == Map.'Map.empty'
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.vertex' (Right 1)))                             == Map.'Map.singleton' 2 Set.'Set.empty'
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 2)))                      == Map.'Map.singleton' 2 (Set.'Set.singleton' 1)
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.edge' (Left 1) (Right 1)))                      == Map.'Map.singleton' 1 (Set.'Set.singleton' 1)
+    -- rightToLeft ('fromGraph' ('Algebra.Graph.edges' [(Left 1, Right 1), (Left 1, Right 2)])) == Map.'Map.fromAscList' [(1, Set.'Set.singleton' 1), (2, Set.'Set.singleton' 1)]
+    -- @
+    rightToLeft :: Map.Map b (Set.Set a)
+} deriving (Eq, Show)
+
+-- | Check that the internal graph representation is consistent, i.e. that all
+-- edges that are present in the 'leftToRight' map are present in the
+-- 'rightToLeft' map.
+-- Complexity: /O(m log m)/
+--
+-- @
+-- consistent ('fromGraph' 'Algebra.Graph.empty')      == True
+-- consistent ('fromGraph' ('Algebra.Graph.vertex' x)) == True
+-- consistent ('fromGraph' ('Algebra.Graph.edge' x y)) == True
+-- consistent ('fromGraph' ('Algebra.Graph.edges' xs)) == True
+-- consistent ('fromGraph' g)          == True
```

Comment:
This property covers all of the properties above, so I suggest to delete them.

Reply:
I replaced them with the brand new primitives.